orchestrator/migrations/helpers.py

```diff
@@ -975,16 +975,16 @@ def convert_resource_type_relations_to_instance_relations(
     conn.execute(
         sa.text(
             """
-            INSERT INTO subscription_instance_relations (parent_id, child_id, order_id, domain_model_attr)
-            WITH node_children AS (
-                SELECT siv.value as subscription_id, siv.subscription_instance_id as child_instance_id, si.product_block_id
+            INSERT INTO subscription_instance_relations (in_use_by_id, dependent_on_id, order_id, domain_model_attr)
+            WITH dependencies AS (
+                SELECT siv.value as subscription_id, siv.subscription_instance_id as in_use_by_instance_id, si.product_block_id
                 FROM subscription_instance_values AS siv
                 left join subscription_instances as si on siv.subscription_instance_id = si.subscription_instance_id
                 WHERE siv.resource_type_id=:resource_type_id
             )
-            SELECT child_instance_id AS parent_id, si.subscription_instance_id AS child_id, '0' AS order_id, :domain_model_attr AS domain_model_attr
+            SELECT in_use_by_instance_id AS in_use_by_id, si.subscription_instance_id AS dependent_on_id, '0' AS order_id, :domain_model_attr AS domain_model_attr
```

Comment:
When the schema changes I think we should add new versions of migration helpers, instead of changing existing ones

Reply:
The moment they migrate, `parent_id` and `child_id` will change and it would immediately break the old helpers.

Reply:
Exactly. IMO we don't want to rewrite queries for old migrations that are already live. Rewriting helpers would do just that :) 

The old helpers will be automatically deprecated as they don't work on the new schema version - let's call it the v2 schema.
The new helpers will _only_ work for schema v2.
So data migrations prior to schema v2 will still work (i.e. when we run tests all migrations are applied in sequence) because they can use the v1 helper, whereas data migrations based on schema v2 can use the v2 helper.

@acidjunk @mrijk what do you think?

Reply:
How would we go about making the v2, for my understanding the core migrations run before any other migrations, so migrations on top of core will break that use the old helpers.

Reply:
Oh indeed, core migrations run first because we can't make them dependent on our internal migrations, forgot about that.

Then my point about our internal data migrations using a different helper based on the state of the schema is also invalid. And indeed all data migrations -including those written prior to the schema change- will need to be changed in retrospect. Now that we have the migration helpers unified in core at least we can do it in a controlled fashion, but it's still a bit iffy IMO. 

Anyway, you're right, carry on :)